lib/Runtime/Library/amd64/JavascriptFunctionA.S

```diff
@@ -2,27 +2,41 @@
 // Copyright (C) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
 //-------------------------------------------------------------------------------------------------------
+
 .intel_syntax noprefix
+#include "unixasmmacros.inc"
 
-#include "pal/inc/unixasmmacros.inc"
-        
 .extern _ZN2Js18JavascriptFunction13DeferredParseEPPNS_14ScriptFunctionE
 
-.global C_FUNC(amd64_CallFunction)
 .global	_ZN2Js18JavascriptFunction20DeferredParsingThunkEPNS_16RecyclableObjectENS_8CallInfoEz
 
 #ifndef __APPLE__
-.type C_FUNC(amd64_CallFunction), @function
 .type _ZN2Js18JavascriptFunction20DeferredParsingThunkEPNS_16RecyclableObjectENS_8CallInfoEz, @function
 #endif
-        
-.text
-C_FUNC(amd64_CallFunction):
+
+//------------------------------------------------------------------------------
+// Invoke
+//
+//      JavascriptMethod(RecyclableObject* function, CallInfo callInfo, ...)
+//
+// with a custom calling convention in order to support JavascriptStackWalker
+// and RUNTIME_ARGUMENTS. JavascriptMethod entry stack layout:
+//
+//      [Return Address]
+//      [function]              == RDI
+//      [callInfo]              == RSI
+//      [arg0]
+//      [arg1]
+//      ...
+//------------------------------------------------------------------------------
+NESTED_ENTRY amd64_CallFunction, _TEXT, NoHandler
         // Var amd64_CallFunction(RecyclableObject *function,
         //	JavascriptMethod entryPoint, CallInfo callInfo, uint argc, Var *argv) ;
 
         // Save existing rbp
-        push rbp
+        push_nonvol_reg rbp
```

Comment:
> When do we use push_nonvol_reg and should we do this for the rest of the callee saves?

It is a macro to push reg and adjust CFA offset accordingly. The next annotation changed CFA relative register to rbp and rbp remains unchanged in this function, thus all the rest push/pop doesn't affect CFA offset and doesn't need any adjustment. That's why all the rest are direct push/pop. (Copied this from a PAL .S file.That's my understanding.)